### DISCUSSION TITLE
Failed to start celery flower because of Package Collision

### DISCUSSION THREAD

**Discussion by Tristan-Chow** [apache/airflow#26451] [Q&A]
So I start airflow celery flower on my local machine and it reports me ''ImportError: cannot import name 'token_urlsafe' from 'secrets' from options.py of flower package, the loading path points to the secrets directory (airflow/secrets/__init__.py) under airflow project.
Any thoughts on that? I do not think change the directory name is a good choice since it changes the code structure, also I already install python secrets package in my own pycharm venv, so I am thinking maybe I can add some codes to change the loading order or something like that?
Thanks for every help!
<img width="1128" alt="image" src="https://user-images.githubusercontent.com/32355665/190858148-f764a6dd-83fe-4368-911c-0fbcf8fc6349.png">

**Comment by Taragolis** [comment]
@Tristan-Chow did you use pip [constraints](https://airflow.apache.org/docs/apache-airflow/stable/installation/installing-from-pypi.html#installation-tools) when you installed airflow and dependencies?

Or you just cloned airflow from github and mark `airflow` directory as a "Sources Root" in PyCharm?

**Reply by Tristan-Chow** [reply]
Yeah I did the latter one, and the venv environment is built by PyCharm itself when importing the project.
Other roles works normally, only celery flower can not be initialized

**Reply by Taragolis** [reply]
You shouldn't add airflow directory itself as directory Sources Root, in this case everything inside this directory will able to import as  separate package. Expected import`from airflow import secrets` unexpected `import secrets`

By default everything in working directory will add in python search path, default working directory in PyCharm - project root.

For more detail you could read [this](https://realpython.com/lessons/module-search-path/).
For check current search path you could run this scratch/snippet

```python
import sys

for ix, pth in enumerate(sys.path, start=1):
    print(f"Order priority {ix}: {pth}")
```
